### PR TITLE
DOCSP-41123: cxn options

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41123>
Staging - https://preview-mongodbrustagir.gatsbyjs.io/kotlin-sync/DOCSP-41123-options/connect/connection-options/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/connect/connection-options.txt [hunk 1]
```diff
@@ -18,4 +18,316 @@ Specify Connection Options
    :keywords: connection string, URI, server, Atlas, settings, configure
```
Comment (mcmorisi) [inline source/connect/connection-options.txt:19]: Suggestion: If we change the ToC depth to 2 here we can have the Connection Options subsections appear in the ToC.

**Review by mcmorisi — APPROVED** [review]
LGTM with a small suggestion!

File: source/connect/connection-options.txt [hunk 2]
```diff
@@ -18,4 +18,316 @@ Specify Connection Options
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can set connection options in a ``MongoClientSettings`` instance by
+using methods from the ``MongoClientSettings.Builder`` class, then passing
+the settings object to the ``MongoClient.create()`` method.
+
+Configuring the connection this way makes it easier to
+change settings at runtime and can help you catch errors at compile time.
+
+The following example shows how to set options when creating a
```
Comment (rozza) [inline source/connect/connection-options.txt:55]: Recommend adding a section just using the MongoClientSettings.builder eg:

```
        MongoClientSettings
            .builder()
            .applyToClusterSettings { it.hosts(listOf(ServerAddress("127.0.0.1"))) }
            .applyToSocketSettings { it.connectTimeout(60000, TimeUnit.MILLISECONDS) }
            .applyToSslSettings { it.enabled(true) }
            .build()
```

Then give the example below mixing a connection string and settings.

**Review by rozza — APPROVED** [review]
LGTM!

**Pull request merged**